var/spack/repos/builtin/packages/gcc/package.py

```diff
@@ -271,6 +271,7 @@ class Gcc(AutotoolsPackage, GNUMirrorPackage):
     patch('glibc-2.31-libsanitizer-2.patch', when='@8.1.0:8.3.0,9.0.0:9.2.0')
     patch('glibc-2.31-libsanitizer-2-gcc-6.patch', when='@5.3.0:5.5.0,6.1.0:6.5.0')
     patch('glibc-2.31-libsanitizer-2-gcc-7.patch', when='@7.1.0:7.5.0')
+    patch('https://gcc.gnu.org/git/?p=gcc.git;a=patch;h=2b40941d23b1570cdd90083b58fa0f66aa58c86e', sha256='b48e48736062e64a6da7cbe7e21a6c1c89422d1f49ef547c73b479a3f3f4935f', when='@:11.1.0')
```

Comment:
Did you try the patch with older versions of GCC? I wanted to double check a couple before merging and this is what I get:
```console
$ spack patch gcc@:10
==> Using cached archive: /home/culpo/PycharmProjects/spack/var/spack/cache/_source-cache/archive/64/64f404c1a650f27fc33da242e1f2df54952e3963a49e06e73f6940f3223ac344.tar.xz
==> Using cached archive: /home/culpo/PycharmProjects/spack/var/spack/cache/_source-cache/archive/b4/b48e48736062e64a6da7cbe7e21a6c1c89422d1f49ef547c73b479a3f3f4935f
The text leading up to this was:
--------------------------
|diff --git a/libsanitizer/sanitizer_common/sanitizer_platform_limits_posix.cc b/libsanitizer/sanitizer_common/sanitizer_platform_limits_posix.cc
|index d823a12190c..e8fce8a0287 100644
|--- a/libsanitizer/sanitizer_common/sanitizer_platform_limits_posix.cc
|+++ b/libsanitizer/sanitizer_common/sanitizer_platform_limits_posix.cc
--------------------------
File to patch: X
X: No such file or directory
Skip this patch? [y] y
3 out of 3 hunks ignored
2 out of 2 hunks FAILED -- saving rejects to file libsanitizer/sanitizer_common/sanitizer_platform_limits_posix.h.rej
==> Patch https://gcc.gnu.org/git/?p=gcc.git;a=patch;h=2b40941d23b1570cdd90083b58fa0f66aa58c86e failed.
==> Error: Command exited with status 1:
'/usr/bin/patch' '-s' '-p' '1' '-i' '/tmp/culpo/spack-stage/spack-stage-ou1_9fpk/?p=gcc.git;a=patch;h=2b40941d23b1570cdd90083b58fa0f66aa58c86e' '-d' '.'
```

Reply:
Yes, gcc 7.5.0 and gcc 8.5.0

Reply:
I think we need to split this patch in two. The issue is that `sanitizer_platform_limits_posix.cc` got renamed at some point to `sanitizer_platform_limits_posix.cpp`. For later GCC you can use this:
```python
patch('https://github.com/gcc-mirror/gcc/commit/745dae5923aba02982563481d75a21595df22ff8.patch', sha256='7686d3b3a7ad0dc3f44a4141067133d2c708ea2cba503f684389e7bbb3bfa81a', when='@:11.1.0')
```
module the lower bound for versions.

Reply:
Hi @alalazo 

I have just pushed the changes taking into account the actual extension for the versions less (greater) or equal than 9.30 (10.1.0), such that `spack patch` works. The only thing is strange is the case with version 9.4.0 which does not work with either of the patches.

Reply:
Hi @alalazo, @michaelkuhn

I have updated the script with the requested changes. I checked with `spack patch`, as well as, compiling some specific versions requiring (or not) either of the patches.